openedx/core/djangoapps/content/learning_sequences/api/processors/special_exams.py

```diff
@@ -54,26 +57,49 @@ def exam_data(self, pruned_course_outline: UserCourseOutlineData) -> SpecialExam
                         continue
 
                     special_exam_attempt_context = None
-                    try:
-                        # Calls into edx_proctoring subsystem to get relevant special exam information.
-                        # This will return None, if (user, course_id, content_id) is not applicable.
-                        special_exam_attempt_context = get_attempt_status_summary(
-                            self.user.id,
-                            str(self.course_key),
-                            str(sequence.usage_key)
-                        )
-                    except ProctoredExamNotFoundException:
-                        log.info(
-                            'No exam found for {sequence_key} in {course_key}'.format(
-                                sequence_key=sequence.usage_key,
-                                course_key=self.course_key
+
+                    # if exam waffle flag enabled, then use exams logic
+                    if exams_ida_enabled(self.course_key):
+                        # add short description based on exam type
+                        if sequence.exam.is_practice_exam:
+                            exam_type = gettext_noop('Practice Exam')
+                        elif sequence.exam.is_proctored_enabled:
+                            exam_type = gettext_noop('Proctored Exam')
+                        elif sequence.exam.is_time_limited:
+                            exam_type = gettext_noop('Timed Exam')
+                        # todo: Do I need this default?
```

Comment:
I'm guessing I don't need a default since the above three are the only possibilities?

Reply:
my standard for this sort of thing would be to have a default and also log something like "using default, considered impossible" because it wouldn't take much data corruption or new work to make it possible again